node/src/components/consensus/protocols/highway.rs

```diff
@@ -582,22 +584,33 @@ where
                     }
                 }
             }
+            Ok(HighwayMessage::RequestLatestState(panorama)) => {
+                let state = self.highway.state();
+
+                let merged_panorama = state.panorama().merge(state, &panorama);
+
+                merged_panorama
+                    .iter_correct_hashes()
+                    .filter_map(|hash| state.wire_unit(hash, *self.highway.instance_id()))
+                    .map(|swu| {
+                        let msg = HighwayMessage::NewVertex(Vertex::Unit(swu));
+                        let serialized_msg =
+                            bincode::serialize(&msg).expect("should serialize message");
+                        ProtocolOutcome::CreatedTargetedMessage(serialized_msg, sender.clone())
+                    })
+                    .collect()
+            }
         }
     }
 
-    fn handle_latest_state_request(&mut self, sender: I) -> Vec<ProtocolOutcome<I, C>> {
-        let state = self.highway.state();
-
-        state
-            .panorama()
-            .iter_correct_hashes()
-            .filter_map(|hash| state.wire_unit(hash, *self.highway.instance_id()))
-            .map(|swu| {
-                let msg = HighwayMessage::NewVertex(Vertex::Unit(swu));
-                let serialized_msg = bincode::serialize(&msg).expect("should serialize message");
-                ProtocolOutcome::CreatedTargetedMessage(serialized_msg, sender.clone())
-            })
-            .collect()
+    fn handle_new_peer(&mut self, peer_id: I) -> Vec<ProtocolOutcome<I, C>> {
```

Comment:
I'm wondering whether we should handle this call only if we have just joined the network. Otherwise, we end up sending those requests every time we have a new peer connected.

Reply:
I think that's a good thing, actually. We'll just make sure we're both as up to date as possible when connecting to other peers.